src/colorizer/ColorizeCanvas.ts

```diff
@@ -152,17 +154,12 @@ export default class ColorizeCanvas {
   private renderer: WebGLRenderer;
   private pickRenderTarget: WebGLRenderTarget;
 
-  protected dataset: Dataset | null;
-  protected track: Track | null;
-  protected canvasResolution: Vector2 | null;
+  protected params: CanvasStateParams | null;
```

Comment:
Replaced a lot of the tracked properties on the left with a single object, `params`.